socketio/src/payload.rs

```diff
@@ -1,31 +1,45 @@
 use bytes::Bytes;
+use serde_json::Value;
 
 /// A type which represents a `payload` in the `socket.io` context.
-/// A payload could either be of the type `Payload::Binary`, which holds
-/// data in the [`Bytes`] type that represents the payload or of the type
-/// `Payload::String` which holds a [`std::string::String`]. The enum is
-/// used for both representing data that's send and data that's received.
+/// The enum is used for both representing data that's send and
+/// data that's received.
 #[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
 pub enum Payload {
     Binary(Bytes),
-    String(String),
+    Json(Value),
+    Multi(Vec<RawPayload>),
```

Comment:
Not sure if this is better, or if this type even should be recursive, another option would be:
```suggestion
    Multi(Vec<Box<Payload>>),
```

Reply:
I tried `Multi(Vec<Payload>)` first time, recursive actually can be convert to valid json, but not what we expected. To not let user do that, `RawPayload` is provided. also let code simple.

Reply:
I think this makes it a bit obvious that restructuring payload is probably a bad idea. What you do here is basically rewrite a limited Json `Value` enum, with support for binary data. I would rather leverage json, and solve this problem differently. I am going to take a look at this tomorrow, but I am not really in favor of such changes, as they present obvious problems.